axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/IntercoServiceImpl.java

```diff
@@ -247,25 +255,31 @@ public Invoice generateIntercoInvoice(Invoice invoice) throws AxelorException {
     PartnerService partnerService = Beans.get(PartnerService.class);
     InvoiceRepository invoiceRepository = Beans.get(InvoiceRepository.class);
     InvoiceService invoiceService = Beans.get(InvoiceService.class);
+
+    boolean isPurchase;
     // set the status
     int generatedOperationTypeSelect;
     int priceListRepositoryType;
     switch (invoice.getOperationTypeSelect()) {
       case InvoiceRepository.OPERATION_TYPE_SUPPLIER_PURCHASE:
         generatedOperationTypeSelect = InvoiceRepository.OPERATION_TYPE_CLIENT_SALE;
         priceListRepositoryType = PriceListRepository.TYPE_SALE;
+        isPurchase = true;
```

Comment:
The variable 'isPurchase' should depend on generatedOperationTypeSelect and not invoice.getOperationTypeSelect(), 
means you have to revert (true to false and false to true) all assignment to this variable.